mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go

```diff
@@ -498,6 +505,9 @@ func resourceGoogleProjectDelete(d *schema.ResourceData, meta interface{}) error
 	if err != nil {
 		return err
 	}
+	if d.Get("deletion_protection").(bool) {
```

Comment:
@slevenick , the logic is that if `deletion_protection` is set to true, it returns error even when `skip_delete` is set to false. It makes sense to me. Can you double check it? Thanks.

Reply:
It would make sense to me if skip_delete took precedence over deletion_protection, but I could be convinced the other way.

It seems reasonable that setting skip_delete = true indicates that you don't want deletion protection, because you wouldn't actually be deleting anything.

Reply:
My original thought is that `deletion_protection` prevents the deletion from Terraform state and sending DELETE API. `skip_delete` only removes resource from Terraform state.

If `skip_delete` takes precedence over `deletion_protection`, is it necessary to add `deletion_protection`? Can we just set the default value of `skip_delete` to true and not adding `deletion_protection`?

Reply:
Please correct me if I'm wrong, but consider the scenario where skip_delete is set and there's no deletion_protection.
In that case if someone goes ahead and tries to change tags field, it will lead to deletion of project from terraform (and ofc skip delete from actual CRM) and then when Terraform tries to recreate the project with the given tags, it will fail with ALREADY_EXISTS error. 
I'm not aware of the context/rationale behind having a skip_delete field but it surely sounds confusing to have it in play alongside the tags immutable field behavior.

Another point to note is that deletion_protection will need to be added to CRM folders resource as well, so it will keep the behavior uniform for folders and projects instead of having separate mechanisms to handle deletion protection.

So overall I'm in favor of having a deletion_protection field instead of relying on skip_delete.

Reply:
Sure, but when we add this field it's not strictly in service of preventing recreation due to tags changing. I'm thinking of a situation where a user wants to use skip_delete as it was originally designed and it seems to me that behavior would take precedence over deletion_protection if it's unspecified.

> and then when Terraform tries to recreate the project with the given tags, it will fail with ALREADY_EXISTS error.

This doesn't seem like a big deal to me, because the project wasn't actually deleted it should be easily importable by the user.

Reply:
I will move the discussion to our team chat room.

Reply:
After discussing with the team, `skip_delete` should take the precedence over `deletion_protection` as Sam suggested. @abd-goog , can you make the change? Thanks.

Reply:
Sure @zli82016 , thanks for discussing with the team. 
Just to confirm what you mean by skip_delete takes precendence, could you please specify expected Terraform behavior when skip_delete is set to true and deletion_protection is set to true?
Should it just go ahead and delete the project from Terraform and not call DeleteProject API, that's it? Or there should be any message based on deletion_protection config as well?

Or in other words are we saying that deletion_protection comes into effect only if skip_delete is set to false?

Reply:
`deletion_protection` comes into effect only if `skip_delete` is set to false.